torch/_inductor/codegen/triton.py

```diff
@@ -1353,16 +1375,23 @@ def load(self, name: str, index: sympy.Expr):
         #   2.1) We are in a reduction loop
         #   2.2) Its not its last use
         #   2.3) This load will not be lifted to the body
+        #
+        is_coalesced = any([i == 1 for i in self.get_strides_of_load(original_index).values()])
         if self.is_broadcasted(original_index):
             ep = ", eviction_policy='evict_last'"
+        elif not is_coalesced:
+            ep = ", eviction_policy='evict_last'"
         elif self.inside_reduction and not self.persistent_reduction:
             if name in self.args.inplace_buffers:
                 names = set(self.args.inplace_buffers[name].other_names)
             else:
                 names = {name}
             last_use = len(names & self.last_usage) > 0
             evict_last = not last_use and ("rmask" in mask or indirect_indexing)
-            ep = ", eviction_policy='evict_last'" if evict_last else ""
+            if evict_last:
+                ep = ", eviction_policy='evict_last'"
+            else:
+                ep = ", eviction_policy='evict_first'"
         else:
             ep = ""
```

Comment:
evict_first ehre as well?

Reply:
Possibly, but I think generally `evict_first` isn't something that should always be thrown in.

Reply:
I think it makes sense to use it given how we structure our kernels, where most of the loads are at the beginning, and then we don't reload them again